src/CoreBluetooth/AdvertisementData.cs

```diff
@@ -0,0 +1,32 @@
+using System;
+using System.Collections.Generic;
+using System.Runtime.InteropServices;
+using System.Linq;
+
+using XamCore.ObjCRuntime;
+using XamCore.Foundation;
+using XamCore.CoreFoundation;
+
+namespace XamCore.CoreBluetooth {
+
+	public partial class AdvertisementData {
+		
+		public IDictionary <CBUUID, NSData> ServiceData {
+			get {
+				using (var nsDict = _ServiceData) {
```

Comment:
Why not just return the native dictionary ? i.e. `NSDictionary<CBUUID, NSData>`

Beside the extra memory and time doing a copy hides the fact that the returned NSDictionary is (likely) not mutable. It can (maybe not here) become unclear if that's a reference or a copy.